tests: Bluetooth: Mesh: Use bsim settings backend

alxelax opened the pull request:
PR adds using nrf52_bsim platform NVMC implementation as well as flash for settings and NVS instead of proprietary custom settings backend.

Andrewpini reviewed (dismissed):
LGTM 👍 One minor comment.

alxelax commented on tests/bsim/bluetooth/mesh/src/test_dfu.c:
```diff
@@ -583,13 +580,11 @@ static void test_dist_dfu_slot_create(void)
 	dist_prov_and_conf(DIST_ADDR);
 
-	for (i = 0; i < CONFIG_BT_MESH_DFU_SLOT_CNT; i++) {
+	for (i = CONFIG_BT_MESH_DFU_SLOT_CNT - 1; i >= 0; i--) {
```
Nop. Neither commit with fix first nor bsim backend first. They will not pass bsim test. This is workaround DFU slot bug when slot sequence depends on settings backend. It should be fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/61389

PavelVPV replied:
I think it should be fine to move this change into separate commit which can be reverted later.

alxelax commented on tests/bsim/bluetooth/mesh/src/mesh_test.c:
```diff
@@ -256,7 +255,10 @@ void bt_mesh_device_setup(const struct bt_mesh_prov *prov, const struct bt_mesh_
 	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
 		LOG_INF("Loading stored settings");
-		settings_load();
+		if (IS_ENABLED(CONFIG_BT_MESH_USES_MBEDTLS_PSA)) {
+			settings_load_subtree("itsemul");
+		}
+		settings_load_subtree("bt");
```
Keys storage area does not belong to Bluetooth. Restoring sequence within the same area is not guaranteed, but keys should be ready at the time when Bluetooth mesh is restored.

PavelVPV replied:
Could you please move this change into a separate commit? This should work independently of the settings backend.

alxelax commented on tests/bsim/bluetooth/mesh/src/psa_its_emul.c:
```diff
@@ -95,7 +95,7 @@ static int itsemul_set(const char *name, size_t len_rd, settings_read_cb read_cb
 }
 
-BT_MESH_SETTINGS_DEFINE(psa_its_emu, "itsemul", itsemul_set);
+SETTINGS_STATIC_HANDLER_DEFINE(psa_its_emu, "itsemul", NULL, itsemul_set, NULL, NULL);
```
Keys storage area does not belong to Bluetooth. Restoring sequence within the same area is not guaranteed, but keys should be ready at the time when Bluetooth mesh is restored.

alxelax committed:
tests: Bluetooth: Mesh: move psa its emulator out of bt subtree

PSA ITS settings area does not belong to Bluetooth.
It should be restored before Bluetooth persistent data.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
tests: Bluetooth: Mesh: use bsim settings backend

Commit adds using BabbleSim's settings backend instead of
proprietary one that is based on POSIX files.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
tests: Bluetooth: Mesh: opposite slot order for bsim settings backend

DFU slot sequence depends on the settings backend.
Commit adapts test after changing bsim settings backend.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

PavelVPV commented on tests/bsim/bluetooth/mesh/_mesh_test.sh:
```diff
@@ -26,20 +26,32 @@ function RunTest(){
 
-  for testid in ${testids[@]} ; do
+  for testid in ${testid_in_order[@]}; do
+    echo "Test id: " $testid
```
This shouldn't be needed?
